plugins/outputs/cloudwatch_logs/cloudwatch_logs.go

```diff
@@ -0,0 +1,434 @@
+#profile = ""
+#shared_credential_file = ""
+
+## Endpoint to make request against, the correct endpoint is automatically
+## determined and this option should only be set if you wish to override the
+## default.
+##   ex: endpoint_url = "http://localhost:8000"
+# endpoint_url = ""
+
+## Cloud watch log group. Must be created in AWS cloudwatch logs upfront!
+## For example, you can specify the name of the k8s cluster here to group logs from all cluster in oine place
+log_group = "my-group-name" 
+
+## Log stream in log group
+## Either log group name or reference to metric attribute, from which it can be parsed:
+## tag:<TAG_NAME> or field:<FIELD_NAME>. If log stream is not exist, it will be created.
+## Since AWS is not automatically delete logs streams with expired logs entries (i.e. empty log stream) 
+## you need to put in place appropriate house-keeping (https://forums.aws.amazon.com/thread.jspa?threadID=178855)
+log_stream = "tag:location"
+
+## Source of log data - metric name
+## specify the name of the metric, from which the log data should be retrieved.
+## I.e., if you  are using docker_log plugin to stream logs from container, then
+## specify log_data_metric_name  = "docker_log"
+log_data_metric_name  = "docker_log"
+
+## Specify from which metric attribute the log data should be retrieved:
+## tag:<TAG_NAME> or field:<FIELD_NAME>.
+## I.e., if you  are using docker_log plugin to stream logs from container, then
+## specify log_data_source  = "field:message" 
+log_data_source  = "field:message"
+`
+
+func (c *CloudWatchLogs) SampleConfig() string {
+	return sampleConfig
+}
+
+func (c *CloudWatchLogs) Description() string {
+	return "Configuration for AWS CloudWatchLogs output."
+}
+
+func (c *CloudWatchLogs) Connect() error {
+	var queryToken *string
+	var dummyToken = "dummy"
+	var logGroupsOutput = &cloudwatchlogs.DescribeLogGroupsOutput{NextToken: &dummyToken}
+	var err error
+
+	if c.LogGroup == "" {
+		return fmt.Errorf("Log group is not set!")
+	}
+
+	if c.LogStream == "" {
+		return fmt.Errorf("Log stream is not set!")
+	}
+
+	if c.LDMetricName == "" {
+		return fmt.Errorf("Log data metrics name is not set!")
+	}
+
+	if c.LDSource == "" {
+		return fmt.Errorf("Log data source is not set!")
+	} else {
+		lsSplitArray := strings.Split(c.LDSource, ":")
+		if len(lsSplitArray) > 1 {
+			if lsSplitArray[0] == "tag" || lsSplitArray[0] == "field" {
+				c.ldKey = lsSplitArray[0]
+				c.ldSource = lsSplitArray[1]
+				log.Printf("D! %s Log data: key '%s', source '%s'...", logId, c.ldKey, c.ldSource)
+			} else {
+				return fmt.Errorf("Log data source is not properly formatted.\n" +
+					"Should be 'tag:<tag_mame>' or 'field:<field_name>'")
+			}
+		} else {
+			return fmt.Errorf("Log data source is not properly formatted, ':' is missed.\n" +
+				"Should be 'tag:<tag_mame>' or 'field:<field_name>'")
+		}
+
+		if c.lsSource == "" {
+			c.lsSource = c.LogStream
+			log.Printf("D! %s Log stream '%s'...", logId, c.lsSource)
+		}
+	}
```

Comment:
Initialization and config checking should be done on the Init() error function, not in the Connect function. Can you move these lines to an Init() function please? I think the rest of the function looks like it belongs in the Connect function.

Reply:
Agree, will fix that